pkg/cmd/server/api/types.go

```diff
@@ -491,6 +491,14 @@ type HTPasswdPasswordIdentityProvider struct {
 
 type LDAPPasswordIdentityProvider struct {
 	api.TypeMeta
+	// ClientConfig is the configuration for the LDAP client connection
+	LDAPClientConfig
+	// Attributes maps LDAP attributes to identities
+	Attributes LDAPAttributeMapping
+}
+
+// LDAPClientConfig holds information for connecting to an LDAP server
+type LDAPClientConfig struct {
 	// URL is an RFC 2255 URL which specifies the LDAP search parameters to use. The syntax of the URL is
 	//    ldap://host:port/basedn?attribute?scope?filter
 	URL string
```

Comment:
I don't really want that field broken down more... from a user's pserspective, it's a succinct, fairly standard way to express that query. With the refactor, it seems like the fields in one struct are being used in different ways, which makes it difficult to explain/document... this struct was built for searching, finding a user, and binding to them, so the URL references search query terms, the bindDN/bindPassword references a "search phase", etc. Now the struct would be used for multiple things, and the doc gets weird... how do you doc the expected format of the URL field when it is used differently for a identity provider and a group syncer?